hydra-module.nix

```diff
@@ -396,17 +396,30 @@ in
 
     # If there is less than a certain amount of free disk space, stop
     # the queue/evaluator to prevent builds from failing or aborting.
+    # Leaves a tag file indicating this reason; if the tag file exists
+    # and disk space is above the threshold + 10GB, the queue/evaluator will be
+    # restarted; starting it if it is already started is not harmful.
     systemd.services.hydra-check-space =
       { script =
           ''
-            if [ $(($(stat -f -c '%a' /nix/store) * $(stat -f -c '%S' /nix/store))) -lt $((${toString cfg.minimumDiskFree} * 1024**3)) ]; then
-                echo "stopping Hydra queue runner due to lack of free space..."
-                systemctl stop hydra-queue-runner
-            fi
-            if [ $(($(stat -f -c '%a' /nix/store) * $(stat -f -c '%S' /nix/store))) -lt $((${toString cfg.minimumDiskFreeEvaluator} * 1024**3)) ]; then
-                echo "stopping Hydra evaluator due to lack of free space..."
-                systemctl stop hydra-evaluator
-            fi
+            spaceleft=$(($(stat -f -c '%a' /nix/store) * $(stat -f -c '%S' /nix/store)))
+            spacestopstart() {  # $1 = job, $2 = min GB free
+              if [ $spaceleft -lt $(($2 * 1024**3)) ]; then
```

Comment:
Can you give useful names to `$1` and `$2` in the function? Also, can you print a message in the "starting" block?

Reply:
Done, @grahamc .